openslides_backend/action/generics/get_meeting_id_mixin.py

```diff
@@ -0,0 +1,19 @@
+from typing import Any, Dict
+
+from ...shared.patterns import FullQualifiedId
+from ..action import Action
+
+
+class GetMeetingIdMixin(Action):
+    """
+    Mixin to easily return the meeting id for update and delete actions.
+    """
+
+    def get_meeting_id(self, instance: Dict[str, Any]) -> int:
+        if instance.get("meeting_id"):
+            return instance["meeting_id"]
+        else:
+            db_instance = self.fetch_model(
```

Comment:
This will fail for organisation, committee, users, ...

Maybe `get_meeting_id` should return `Optional[int]`, so the generic code calling `get_meeting_id` can return an error. This method must be overwritten by custom logic for non-meeting models.

For users: The temporary-user thing is dangerous since you would get a meeting id from the user model! I'm unsure if this could be benefitial or fatal ;).

Reply:
> For users: The temporary-user thing is dangerous since you would get a meeting id from the user model! I'm unsure if this could be benefitial or fatal ;).
I suspect it is more fatal than benefitial: It's really not that what you expect to get (my yesterdays experience!)

Reply:
Why do you implement this as mixin? Wouldn't it be easier to implement a method in Action class and overwrite it, where needed? Returning `Optional[int]` is a good idea.

Reply:
> For users: The temporary-user thing is dangerous since you would get a meeting id from the user model! I'm unsure if this could be benefitial or fatal ;).

This is fine: The use case of this method is to check to which meeting this models belongs to. In case of temporary users, this is the `meeting_id`.


> This will fail for organisation, committee, users, ...

Yes, this is okay. All models/action which are not inside a meeting must override the permission check anyway, so this method never gets called.

> Why do you implement this as mixin? Wouldn't it be easier to implement a method in Action class and overwrite it, where needed?

I don't like putting code into the base class which only works for a part of the subclasses. It seems cleaner to add it as an abstract method and override it via a mixin, if multiple subclasses need the same logic. But ultimately this comes down to personal preference I guess, or do you have any arguments for one or the other?

Reply:
> so this method never gets called.

You should fail hard if it gets called accidentally. So if someone forgets it he should get a good error.

Reply:
It currently does: if this is called on a model which does not have a `meeting_id`, it will fail in line 19 with an `AttributeError`. Which is okay in this case, I think, since it indicates the the programmer fucked up and it should never happen in any code that the client calls (or our CI did not work properly XOR there were absolutely no tests for this action, which are both improbable enough for me).

Reply:
> I don't like putting code into the base class which only works for a part of the subclasses. It seems cleaner to add it as an abstract method and override it via a mixin, if multiple subclasses need the same logic. But ultimately this comes down to personal preference I guess, or do you have any arguments for one or the other?
May be it's only personal preference, but with KISS it's more pythonic: A method to overwrite instead of a mixin class is IMO simper and more stupid;-)

Reply:
Sure, I can change it if you like it better